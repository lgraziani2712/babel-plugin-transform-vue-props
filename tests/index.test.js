'use strict';

const { transformFileSync } = require('@babel/core');

const transformVueProps = require('../src');

function transformFile(fileName) {
  return transformFileSync(`${__dirname}/files/${fileName}.js`, {
    babelrc: false,
    plugins: [transformVueProps],
  }).code;
}

describe('babel-plugin-transform-vue-props', () => {
  describe('Normal props', () => {
    test('Without defaults', () => {
      expect(transformFile('normal-without-defaults')).toMatchSnapshot();
    });
    test('With defaults', () => {
      expect(transformFile('normal-with-defaults')).toMatchSnapshot();
    });
  });

  describe('Vue-types props', () => {
    test('Without defaults', () => {
      expect(transformFile('vue-types-without-defaults')).toMatchSnapshot();
    });
    test('With defaults', () => {
      expect(transformFile('vue-types-with-defaults')).toMatchSnapshot();
    });
  });

  describe('Mixed props', () => {
    test('Without defaults', () => {
      expect(transformFile('mixed-without-defaults')).toMatchSnapshot();
    });
    test('With defaults', () => {
      expect(transformFile('mixed-with-defaults')).toMatchSnapshot();
    });
  });
});
