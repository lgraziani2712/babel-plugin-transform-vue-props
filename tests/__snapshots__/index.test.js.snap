// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-transform-vue-props Mixed props With defaults 1`] = `
"export default {
  props: {
    name: {
      default: 'lolo'
    },
    value: null
  }
};"
`;

exports[`babel-plugin-transform-vue-props Mixed props Without defaults 1`] = `
"export default {
  props: [\\"name\\", \\"value\\"]
};"
`;

exports[`babel-plugin-transform-vue-props Normal props With defaults 1`] = `
"export default {
  props: {
    name: {
      default: 'lolo'
    },
    value: {
      default: 1
    }
  }
};"
`;

exports[`babel-plugin-transform-vue-props Normal props Without defaults 1`] = `
"export default {
  props: [\\"name\\", \\"value\\"]
};"
`;

exports[`babel-plugin-transform-vue-props Vue-types props With defaults 1`] = `
"export default {
  props: {
    name: {
      default: 'lolo'
    },
    value: null
  }
};"
`;

exports[`babel-plugin-transform-vue-props Vue-types props Without defaults 1`] = `
"export default {
  props: [\\"name\\", \\"value\\"]
};"
`;
